lib/twingly/url_cache.rb

```diff
@@ -5,26 +5,38 @@
 
 module Twingly
   class UrlCache
+    class Error < StandardError; end
+    class ServerError < Error; end
+
     attr_reader :ttl
 
     CACHE_VALUE = ""
 
     def initialize(ttl: 0)
-      @cache = Dalli::Client.new(servers, options)
+      @cache = with_memcached_exception_handling do
+        Dalli::Client.new(servers, options)
+      end
+
       @ttl = ttl
     end
 
     def cache!(url)
       key = key_for(url)
-      Retryable.retryable(tries: 3, on: Dalli::RingError) do
-        !!@cache.set(key, CACHE_VALUE, ttl, raw: true)
+
+      with_memcached_exception_handling do
+        Retryable.retryable(tries: 3, on: Dalli::RingError) do
```

Comment:
I think we should use `retryable`'s block rather than introducing yet another level of blocks. It seems like the exception is yieled to the block, see for instance: https://github.com/nfedyashev/retryable#matching-error-messages

Reply:
Also, looking at https://github.com/petergoldstein/dalli/blob/b50bb8245758a722f9525d44f05136cd3a0dac43/lib/dalli.rb#L12-L23 there might be more things "retryable" (`NetworkError` specifically)

Reply:
> I think we should use `retryable`'s block rather than introducing yet another level of blocks. It seems like the exception is yieled to the block, see for instance: https://github.com/nfedyashev/retryable#matching-error-messages

Smart, I'll take a look at that.

> Also, looking at https://github.com/petergoldstein/dalli/blob/b50bb8245758a722f9525d44f05136cd3a0dac43/lib/dalli.rb#L12-L23 there might be more things "retryable" (`NetworkError` specifically)

That I noticed, but we have never seen the NetworkError before (couldn't find any issues regarding that anyway), so I just ignored it.

Reply:
Some improvements in 9a1097fd4a6837fcdf97658df46c5ff941399563